src/ompl/tools/lightning/DynamicTimeWarp.h

```diff
@@ -0,0 +1,117 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dave Coleman
+   Desc:   Compute Dynamic Time Warping
+*/
+
+#ifndef OMPL_TOOLS_LIGHTNING_DYNAMIC_TIME_WARP_
+#define OMPL_TOOLS_LIGHTNING_DYNAMIC_TIME_WARP_
+
+#include <ompl/geometric/PathGeometric.h>
+#include <ompl/base/SpaceInformation.h>
+
+namespace ompl
+{
+namespace tools
+{
+
+namespace og = ompl::geometric;
+namespace ob = ompl::base;
+//namespace ot = ompl::tools;
+
+/// @cond IGNORE
+OMPL_CLASS_FORWARD(DynamicTimeWarp);
+/// @endcond
+
+/** \class ompl::geometric::DynamicTimeWarpPtr
+    \brief A boost shared pointer wrapper for ompl::tools::DynamicTimeWarp */
+
+class DynamicTimeWarp
+{
+public:
+    explicit DynamicTimeWarp(const base::SpaceInformationPtr &si);
+
+    /**
+     * \brief Use Dynamic Timewarping to score two paths
+     * \param path1
+     * \param path2
+     * \return score
+     */
+    double calcDTWDistance(const og::PathGeometric &path1, const og::PathGeometric &path2 ) const;
+
+    /**
+     * \brief Use dynamic time warping to compare the similarity of two paths
+     *        Note: this will not interpolate either of the paths
+     *        Note: before calling this function you might want to reverse one of the paths so that their
+     *        start and goals are property aligned (and match better)
+     * \param path1 - const, will not interpolate
+     * \param path2 - const, will not interpolate
+     * \return score
+     */
+    double getPathsScoreConst(const og::PathGeometric &path1, const og::PathGeometric &path2);
+
+    /**
+     * \brief Use dynamic time warping to compare the similarity of two paths
+     *        Note: this will interpolate the second path and it returns the change by reference
+     *        Note: before calling this function you might want to reverse one of the paths so that their
+     *        start and goals are property aligned (and match better)
+     * \param path1 - const, will not interpolate
+     * \param path2 - will interpolate
+     * \return score
+     */
+    double getPathsScoreHalfConst(const og::PathGeometric &path1, og::PathGeometric &path2);
```

Comment:
I've removed the redundant functions and now just have one that takes two const paths - ``getPathScore()``